### PR TITLE
Yelp crawler 009

### PR DESCRIPTION
**Pull request by mjkoo711**

### CODE REVIEW

File: back-end/yelpCrawler.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+            attr_name, attr_content = "", ""
+            for dataidx in info.children:
+                if dataidx.name == "dt": #tag_name == dt
+                    attr_name = dataidx.string.strip() # strip function removes in bracket.
+                if dataidx.name == "dd": 
+                    attr_content = dataidx.string.strip()
+            moreinfo_data[attr_name] = attr_content
+
+        data['attributes'] = moreinfo_data
+        if data['attributes'].get('Wi-Fi')==wifi and data['attributes'].get('Parking') :
```
Comment (khc196) [inline back-end/yelpCrawler.py]: Any condition doesn't need for Parking?

Comment (mjkoo711) [reply back-end/yelpCrawler.py]: you said to me, if parking key is not exist, there is no parking area. So I did!

File: back-end/yelpCrawler.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        business_id (str): The ID of the business to query.
+    Returns:
+        dict: The JSON response from the request.
+    """
+    business_path = BUSINESS_PATH + business_id
+
+    return request(API_HOST, business_path, api_key)
+
+# Define default place : wework civic, default wifi='free', default parking='street'
+def getYelpData(lat=37.778264, long=-122.411843, wifi="Free", parking="Street"):
```
Comment (khc196) [inline back-end/yelpCrawler.py]: The 'parking' is unused. I want to use 'getYelpData' method as the following screenshot. 
image

Comment (mjkoo711) [reply back-end/yelpCrawler.py]: OK ! I'll try it!

File: back-end/app.py [hunk 3]
```diff
@@ -1,107 +1,113 @@
...
+"""
+from __future__ import print_function
+
+import json
+import os
+
+from flask import Flask, request, make_response
+
+import uuid
+import redis
```
Comment (shberry) [inline back-end/app.py:19]: Is it possible to group imports together?

File: back-end/app.py [hunk 4]
```diff
@@ -1,107 +1,113 @@
...
+    req = request.get_json(silent=True, force=True)
+    res = processWebhookRequest(req)
+
+    # JSON Encoding
+    res = json.dumps(res, indent=VAL_INDENT)
+
+    r = make_response(res)
+    r.headers['Content-Type'] = 'application/json'
+    return r
+
```
Comment (shberry) [inline back-end/app.py:44]: remove extra line

File: back-end/app.py [hunk 5]
```diff
@@ -1,107 +1,113 @@
...
+    data = callCrawler(req)
+    res = makeWebhookResult(data)
+    return res
+
+def makeQuery(req):
+    result = req.get("result")
+    parameters = result.get("parameters")
+    parameter_atmosphere = parameters.get("atmosphere")
+    parameter_facility = parameters.get("mapsort")
+    parameter_open = parameters.get("open")
```
Comment (shberry) [inline back-end/app.py:65]: line 61-62 is repeated in the below def. Can you refactor this out?

File: back-end/app.py [hunk 6]
```diff
@@ -1,107 +1,113 @@
...
+        response=data,
+        status=200,
+        mimetype='application/json'
+    )
+    return response
+
+if __name__ == '__main__':
+    port = int(os.getenv('PORT', 5000))
+
+    print("Starting app on port %d" % port)
```
Comment (shberry) [inline back-end/app.py:111]: Should this be part of the production code? If it's just for debugging, please remove.

File: back-end/yelpCrawler.py [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+    url_params = url_params or {}
+    url = '{0}{1}'.format(host, quote(path.encode('utf8')))
+    headers = {
+        'Authorization': 'Bearer %s' % api_key,
+    }
+    response = requests.request('GET', url, headers=headers, params=url_params)
+
+    return response.json()
+
+#def search(api_key, term, latitude, longitude, offset):
```
Comment (shberry) [inline back-end/yelpCrawler.py:64]: Can this be removed?

File: back-end/yelpCrawler.py [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+    Args:
+        business_id (str): The ID of the business to query.
+    Returns:
+        dict: The JSON response from the request.
+    """
+    business_path = BUSINESS_PATH + business_id
+
+    return request(API_HOST, business_path, api_key)
+
+# Define default place : wework civic, default wifi='free', default parking='street'
```
Comment (shberry) [inline back-end/yelpCrawler.py:96]: Same here, and below.

File: back-end/yelpCrawler.py [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+# def getYelpData(lat=37.778264, longi=-122.411843, wifi="Free", parking="Street"):
+def getYelpData(wifi="Free", parking="Street"):
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('-s', '--term', dest='term', default=DEFAULT_TERM,
+                        type=str, help='Search term (default: %(default)s)')
+    parser.add_argument('-l', '--location', dest='location', default=DEFAULT_LOCATION,
+                        type=str, help='Search term (default: %(default)s)')
+    #parser.add_argument('-la', '--longitude', dest='longitude', default=LONGITUDE,
```
Comment (shberry) [inline back-end/yelpCrawler.py:105]: Please remove all commented lines. If some of these are necessary, please add a comment and justify your reason.

File: back-end/yelpCrawler.py [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+    #                    type=float, help='Search longitude (default: %(default)f)')
+    #parser.add_argument('-lo', '--latitude', dest='latitude', default=LATITUDE,
+    #                    type=float, help='Search latitude (default: %(default)f)')
+
+    input_values = parser.parse_args()
+    total_data_list = []
+    search_limit = 50
+
+    for i in range(20):
```
Comment (shberry) [inline back-end/yelpCrawler.py]: can we use something else than 'i'? To be more meaningful?

File: back-end/yelpCrawler.py [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+                        attr_content = dataidx.string.strip()
+                moreinfo_data[attr_name] = attr_content
+            data['attributes'] = moreinfo_data
+            if data['attributes'].get('Wi-Fi') == wifi and data['attributes'].get('Parking'):
+                datalist.append(data)
+        total_data_list = total_data_list + datalist
+
+    return json.dumps(total_data_list, indent=4)
+
+
```
Comment (shberry) [inline back-end/yelpCrawler.py:201]: remove extra line

File: back-end/yelpCrawler.py [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+                )
+            )
+
+        # Merge loadJsonFile.py in this file
+        datainfo = businessList
+
+        datalist = []
+        for item in datainfo:
+            data = dict()
+            if(item.get('name')): data['name'] = item['name']
```
Comment (shberry) [inline back-end/yelpCrawler.py:138]: Usually it's not recommended to repeatedly use hardcoded string (too many 'name', 'location', and so on). Can you introduce a variable that holds the string, and use that?

**Review by shberry — COMMENTED** [review]
Thanks! PTAL

**Pull request merged**